ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js

```diff
@@ -102,13 +102,33 @@ Draw.PolygonView = Marionette.View.extend({
   },
 
   modelToPolygon(model) {
-    const coords = model.get('polygon')
+    let coords = model.get('polygon')
     if (
       coords === undefined ||
       validateGeo('polygon', JSON.stringify(coords)).error
     ) {
       return
     }
+    if (model.get('bbox')) {
```

Comment:
won't `model.get('bbox')` return true even when it's not a bbox? since the model preserves values on it even when you change your geo (i.e. if you draw a line and then draw a polygon, the model will have both the line and the polygon saved). 

I think we'd just need to test that if you draw and search with a bbox, and then try to draw/search with a polygon, the map looks as expected

Reply:
Because this model's `bbox` is set using the filter's geojson instead of the original location-old model, there won't be a bounding box present if that's not the current mode you're on. There was an issue though when switching from bounding box to polygon because location-old's `setBBox` function was "clearing" the bounding box with `"NaN,NaN,NaN,NaN"`. That caused this line to evaluate to true, resulting in an inability to complete a polygon drawing after having a bounding box. The fix for that is [here](https://github.com/codice/ddf/pull/5885/files#diff-6b0667d4ebab4a516f138f96863e72ddR575)